File: algorithms/007.Reverse_Integer/Reverse_Integer.py
class Solution:
    def reverse(self, x):
        ans = 0
        if x >= 0:
            while x:
                ans = ans * 10 + x % 10
                x = int(x/10)
            return ans if ans <= 2147483647 else 0  # Handle overflow.
        else:
            return -self.reverse(-x)

File: algorithms/007.Reverse_Integer/Reverse_Integer.java
public class Solution {
    public int reverse(int x) {
        int rst = 0;
        
        while(x != 0) {
            int next_rst = rst * 10 + x % 10;
            x = x / 10;
            if(next_rst/10 != rst) {
                rst  = 0;
                break;
            }
            rst = next_rst;
        }
        return rst;
    }
}